Core/include/Acts/EventData/VectorTrackContainer.hpp

```diff
@@ -158,10 +156,27 @@ class VectorTrackContainerBase {
     return m_referenceSurfaces[itrack].get();
   }
 
+  ParticleHypothesis particleHypothesis_impl(IndexType itrack) const {
+    return m_particleHypothesis[itrack];
+  }
+
   std::size_t size_impl() const {
     assert(checkConsistency());
     return m_tipIndex.size();
   }
+
+  // @TODO: Reconsider return type, by-value -> expensive
+  // Could be generic iterator pair, possibly driven by iterator adapter that
+  // unpacks from unordered_map dynamically
+  std::vector<Acts::HashedString> dynamicKeys_impl() const {
```

Comment:
Can we assume this can be pre-calculated and returned as a const ref to a vector?

Reply:
I switched this to a const ref now.